### ISSUE TITLE
Helm Chart for Gateway does not Support Private Registry

### ISSUE THREAD

**Issue by edwardwu-hkjc** [istio/istio#51387]
(This is used to request new product features, please visit <https://github.com/istio/istio/discussions> for questions on using Istio)

**Describe the feature request**
Support specifying Private Registry in Helm Chart for Gateway

**Describe alternatives you've considered**
Tested populating values.yaml with `global.hub="registry url"`

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Dual Stack
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**
Trying to install istio/gateway helm chart in a k8s cluster with no access to docker hub but has a private container registry (e.g. Azure Container Registry)

**Comment by nmnellis** [comment]
Can you please provide an example values.yaml and helm install command?

**Comment by edwardwu-hkjc** [comment]
> Can you please provide an example values.yaml and helm install command?

Hi, I've used the below helm install command and values.yaml. Thanks.

`helm install istio-gateway istio/gateway -n istio-gateway --values istio-gateway-values.yaml`

```yaml
---
defaults:
  global:
    hub: "<example>.azurecr.io/istio"
# other values below, they are working fine
  service:
    annotations:
      service.beta.kubernetes.io/azure-load-balancer-internal: "true"
    loadBalancerIP: "<Private IP Address>" 
```

**Comment by nmnellis** [comment]
this is because gateways are deployed via webhook injection now. Istiod is responsible for setting the image and hub

**Comment by bhat39a** [comment]
This can be controlled using istiod. If you are using ecr you can use pull through cache ecr repo to get the docker images and in your helm codebase add the below setting.

resource "helm_release" "istiod" {
  repository       = "https://istio-release.storage.googleapis.com/charts"
  chart            = "istiod"
  name             = "istiod"
  namespace        = "istio-system"
  create_namespace = false
  depends_on       = [helm_release.istio_base]

  set {
    name  = "meshConfig.outboundTrafficPolicy.mode"
    value = "REGISTRY_ONLY"
  }

  set {
    name  = "meshConfig.accessLogFile"
    value = "/dev/stdout"
  }

  set {
    name  = "pilot.nodeSelector.NodeType"
    value = "${local.name_prefix}-k8s_static_ng"
  }


  set {
    name  = "global.hub"
    value = "xxxxx.dkr.ecr.eu-west-1.amazonaws.com/docker-hub/istio"
  }

}

**Comment by Sabo-kun** [comment]
Hello!

I have to open this issue again, because we noticed an error while installing an ingress gateway with this setup:
- latest istio version available in helm hub (1.26.2) with istio base, istio cni and istiod installed
- kyverno with cluster policy which limits ephemeralContainers/initContainers/containers to only a given registry 123.dkr.ecr.eu-west-1.amazonaws.com

Among the the values passed to istiod helm installation we set:
`
global:
  hub: 123.dkr.ecr.eu-west-1.amazonaws.com/docker-hub/istio
`

If I install the gateway chart with this command:
helm install istio-ingressgateway istio/gateway -n myns -f values.yaml

and the following values:
`
labels:
  istio: "ingressgateway"

securityContext:
  seccompProfile:
    type: RuntimeDefault

service:
  type: NodePort
`

I see the following error during the validation:
`
Error: INSTALLATION FAILED: 1 error occurred:
       * admission webhook "validate.kyverno.svc-fail" denied the request:  

resource Deployment/beats/ingress-gateway was blocked due to the following policies  

restrict-image-registries:
 autogen-validate-registries: 'validation error: Unknown image registry: [{"image":"auto","imagePullPolicy":"Always","name":"istio-proxy","ports":[{"contain
erPort":15090,"name":"http-envoy-prom","protocol":"TCP"}],"resources":{"limits":{"cpu":"2","memory":"1Gi"},"requests":{"cpu":"100m","memory":"128Mi"}},"secur
ityContext":{"allowPrivilegeEscalation":false,"capabilities":{"drop":["ALL"]},"privileged":false,"readOnlyRootFilesystem":true,"runAsGroup":1337,"runAsNonRoo
t":true,"runAsUser":1337},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File"}].
   rule autogen-validate-registries failed at path /spec/template/spec/containers/0/image/'
`

so, the fact that the image is set to:
image: auto
in the helm chart and then changed by a mutating webhook (as @nmnellis reported) is causing the validation fail somehow (even though mutating webhooks should run BEFORE the validation ones https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/ ).

Is there any setting that I am missing to configure the mutation properly?

**Comment by dminca** [comment]
Hey, here's a nice learning I'd like to share with all of you struggling with this one.

Setting this in the values of `istio/istiod` works

```yml
global:
  hub: your-private-registry/istio
```

> [!CAUTION]
> The image will NOT update on the fly if you already have a **running** `Deployment/istio-eastwest-gateway`

> [!TIP]
> Simply trigger a rolling-restart and the new image will be created WITH your desired container registry injected.
